Task2

AleksandrKalinin opened the pull request:
Использовал gulp
1.Html страницы находятся в корне проекта
2.Исходные файлы стилей лежат в папке src/sass, скомпилированные стили в папке build/css, изображения также в папке build

akhramtsou-dev commented:
Касательно описания к Pull Request'ам.

Зачастую разработчики написав четкие лаконичные коммит сообщения считают это описание и оставляют его пустым. Но пустым описание оставлять не нужно.
Описание должно содержать в себе описание основных элементов которые были разработаны/отрефакторены/протестированы и так далее.

Что использовано и где что лежит можно добавлять после описания как follow-up комментарий.
Также предпочтительно объяснять решение в Jira комментарии. Например, для сборки решения использовал Gulp. Использовал SCSS препроцессор для CSS потому и потому.

Пример, того как могло бы выглядеть описание:
- добавить HTML markup для таких-то страниц
- стилизовать страницы с использованием BEM
- стилизация под мобильные устройства
- определить общие стили в отдельной папке
- начальная установка Gulp (плагины и т.д.)

akhramtsou-dev commented:
Functional issues
1 Do not matches dimensions/spacing (ex. sidebar outer spacing (left 32px (34px correct), top 50px (60px correct), mobile, etc). Ideally should be pixel perfect.
2 Containers are not scrollable in galleries (small, medium)
3 Fixed header and search on scroll
4 Absolute positioning (Subscribe CTA in gallery header)
5 Animations outside of scope (Sidebar menu items on click, gallery item on hover)
6 Menu items on mobile (Home page)

Tech issues
1 Split long SCSS files into multiple files (by feature, by page)
See docs [@use](https://sass-lang.com/documentation/at-rules/use) (or [@import](https://sass-lang.com/documentation/at-rules/import))
2 Mixed classes in selectors
`.gallery-small .gallery-small__wrapper .gallery-small__item .gallery-small__description .small-item__title`
No styles for `.small-item`.
3 Reuse Typography styles
Font sizing for sidebar `.subscriptions-item__text and .categories-item__text`
4 Semantic HTML (<nav> for sidebar, etc.)
5 Accessibility concerns (img alt, role)
6 BEM everywhere? (header block)
7 Button regular with a element (Subscribe CTA, gallery navigation item)
8 Decision on headings (h4, h5) based on what?
9 Icons as images (use more flexible format, SVG for example)
10 Presentational icons (search icon in search)

akhramtsou-dev commented on gulpfile.js:
```diff
@@ -0,0 +1,98 @@
+			cascade: false
+		}))
+
+/*		.pipe(cleanCSS({
```
Clean up unnecessary code.

akhramtsou-dev commented on gulpfile.js:
```diff
@@ -0,0 +1,98 @@
+]
+
+const jsFiles = [
+	'./src/js/main.js'
```
Do you need script? It looks like it's empty.

akhramtsou-dev commented on package.json:
```diff
@@ -0,0 +1,33 @@
+  "description": "",
+  "main": "index.js",
+  "dependencies": {
+    "browserlist": "^1.0.1",
```
Are those dependencies should be listed here? Consider putting them in `devDependencies`?

akhramtsou-dev commented:
It would be nice to have README with short description of task and solution details, tech stack.

AleksandrKalinin commented:
1.Implemented scroll in gallery component;
2.Created a separate sidebar menu for moblie devices instead of restyling desktop version of it;
3.Removed unnecessary animations;
4.Made main header fixed;
5.Implemented relative positioning instead of absolulte for several elements;
6.Made use of semantic Html tags(<nav> in particular);
7.Changed several HTML tags for more appropriate ones (<span> instead of <a>)
7.Added alt attribute to images
8.Used SVG format for icons
9.Used background properry for search icon in search bar instead of inserting <img> element
10..Removed unnecessary code
11.Splitted main SASS file to modules for every major element, made style files more readable
12.General markup and styling fixes(spacing, padding, font-size, etc.)

akhramtsou-dev commented:
@AleksandrKalinin, latest changes look good.
Here some notes on several items:
- `alt` for images added with text that do not describe images
- SVG icons look good (explore ways to embed icons without <img> element, it can be more flexible and allow to control some attributes with CSS like `fill: currentColor`, etc)

I still seeing some issue with mobile version of Hope page not matching design (it't not a small spacing issue). I see incorrect margin in grid, mess in header.

akhramtsou-dev commented:
I have a few more issues to fix.

Functional issues
...
7 Review styling for Channel and Video pages on mobile

Tech issues
11 `.content-player` with fixed units
12 Using semantic tags for building Recommended Channels block (see [example](https://stackoverflow.com/a/7295013))
13 `.videoplayer-playlist` without vertical scroll
14 Not good naming, example `.videoplayer-description__thumbnail`
15 It’s possible to reuse classes for avatars (for example channel avatar on Video and Home pages)
16 Styles for one element in separate places `.switch-text`

AleksandrKalinin commented:
Applied changes:
1. Fixed styling for mobile devices
2. Changed alt attributes to more informative
3. Added scrollbar to playlist on video page
4. Switched to semantic tags in recommended component
5. Made use of relative units
6. Made profile images reusable
7. Removed .switch-text duplicates
8. Changed to more appropriate class naming instead of .videoplayer-description__thumbnail
